app/models/user_access.rb

```diff
@@ -201,6 +211,17 @@ def grant_access(new_user, selected_facility_ids)
 
   private
 
+  def has_access?(resource)
+    case resource
+    when Facility
+      user.accesses.exists?(resource: resource) || has_access?(resource.facility_group)
+    when FacilityGroup
+      user.accesses.exists?(resource: resource) || has_access?(resource.organization)
+    when Organization
+      user.accesses.exists?(resource: resource)
+    end
+  end
```

Comment:
check for direct access, and then works up the chain to see if a parent FG or Org grants access